runtime/src/accounts_db.rs

```diff
@@ -573,12 +576,59 @@ impl AccountsDB {
         let mut reclaims = Vec::new();
         let mut accounts_index = self.accounts_index.write().unwrap();
         for purge in &purges {
-            reclaims.extend(accounts_index.purge(purge));
+            reclaims.extend(accounts_index.would_purge(purge));
         }
         let last_root = accounts_index.last_root;
+
+        // Calculate store counts as if everything was purged
+        // Then purge if we can
+        let mut store_counts: HashMap<AppendVecId, usize> = HashMap::new();
+        let storage = self.storage.read().unwrap();
+        for (slot_id, account_info) in &reclaims {
+            if let Some(slot_storage) = storage.0.get(&slot_id) {
```

Comment:
Make this just a `wrap`s because something is very broken if we get `None` for `slot_storage` or `store_count`. Or at least add `warn!`s by adding `else` clauses.

Reply:
Added `unwrap()` since these should always be present.